### ISSUE TITLE
snapclient not working on Ubuntu Desktop 19.10

### ISSUE THREAD

**Issue by PD75** [badaix/snapcast#567]
I have been looking through issues people have and I have not been able to get sound from my NUC installed with Ubunutu Desktop 19.10. It work perfectly out of the box in my Pi4 with Raspbian.

The snapclient versions tried 15.0 and 18.1 (which I also run on the Pi4 but there I don't have pulseaudio).

I tried to shutdown pulseaudio did not help, desktop sound still works though.

I am lost where to start...

**Comment by badaix** [comment]
what happens when you start snapclient on a terminal?

**Comment by PD75** [comment]
Says something about failed to get chunk. Not sure how to interpret. All devices are on 1 Gbit Ethernet.

pd@LinuxMedia:~$ snapclient
2020-03-20 15-42-41 [Info] (Browser) CACHE_EXHAUSTED
2020-03-20 15-42-41 [Info] (Browser) NEW: service 'Snapcast' of type '_snapcast._tcp' in domain 'local'
2020-03-20 15-42-41 [Info] Service 'Snapcast' of type '_snapcast._tcp' in domain 'local':
2020-03-20 15-42-41 [Info]      WinNasLinux.local:1704 (192.168.1.110)
2020-03-20 15-42-41 [Info] Found server 192.168.1.110:1704
2020-03-20 15-42-41 [Info] Latency: 0
2020-03-20 15-42-41 [Notice] Connected to 192.168.1.110
2020-03-20 15-42-41 [Info] My MAC: "1c:69:7a:07:8d:a0", socket: 6
2020-03-20 15-42-41 [Info] ServerSettings - buffer: 1000, latency: 0, volume: 100, muted: 0
metadata:{"STREAM":"default"}
2020-03-20 15-42-41 [Info] Codec: flac
2020-03-20 15-42-41 [Notice] sampleformat: 48000:16:2
2020-03-20 15-42-41 [Info] frames: 1440
2020-03-20 15-42-41 [Info] Failed to get chunk
2020-03-20 15-42-41 [Info] age > 200: -986
2020-03-20 15-42-41 [Info] sleep < -bufferDuration/2: -986 < -15, sleep: -986
2020-03-20 15-42-41 [Info] sleep < -bufferDuration/2: -986 < -15, sleep: -955
2020-03-20 15-42-41 [Info] sleep < -bufferDuration/2: -955 < -15, sleep: -925
2020-03-20 15-42-41 [Info] sleep < -bufferDuration/2: -925 < -15, sleep: -895
2020-03-20 15-42-41 [Info] sleep < -bufferDuration/2: -895 < -15, sleep: -865
2020-03-20 15-42-41 [Info] diff to server [ms]: 271.17
2020-03-20 15-42-43 [Info] sleep < -bufferDuration/2: -865 < -15, sleep: 1084
2020-03-20 15-42-43 [Info] Sleep 1084, age: 1084, bufferDuration: 30
2020-03-20 15-42-43 [Info] Chunk: 10846 10846   10846   10846   1       149
2020-03-20 15-42-43 [Info] Failed to get chunk
2020-03-20 15-42-43 [Info] sleep > bufferDuration/2: 1084 > 15
2020-03-20 15-42-43 [Info] sleep > chunkDuration: 1084 > 24, chunks: 79, out: 149, needed: 30
2020-03-20 15-42-43 [Info] sleep > chunkDuration: 1054 > 24, chunks: 78, out: 149, needed: 30

**Comment by badaix** [comment]
Any reason for not using the v0.19?
are you using WiFi?

**Comment by PD75** [comment]
All devices are 1Gbit Ethernet with full duplex all switches also, I get 7-800Mbps on Internet from all devices and single digit to low double digit ms latency.

I had major problems when I moved from 15 to 18.0 , with dropouts in sound every few seconds making the setup unusable, so I reverted back to 15. I think my Pi4 updated to 18.1 during regular updates The server is still 15. That's why I tested both 15 and 18.1 on the Intel Ubuntu system,

I am thinking to test again to upgrade to 18.1 or 19, but that is a project for tomorrow. I want to backup the hard-drives images since I don't feel too confident where all of the settings are. 

Will get back with the result.

**Comment by badaix** [comment]
Hmm, your current problem (from which you pasted the logs) looks unrelated to the drops. 
But anyway, I highly recommend to update the server, as the regular dropouts should be solved there. From client point of view the 0.19 introduces a better and actually simpler synchronization without such "sleep" messages that are even for me hard to understand.

**Comment by PD75** [comment]
So, now upgraded to 0.19 on Server, Linux NUC and Pi4, working flawlessly also on Pi4 and my phone, thanks for the advice. 

Due to collision at home I had to move my stuff, and found there was a head phone jack on the NUC (I have not used analog audio this millenium, but I plugged my old PC speakers in and it works there, but not HDMI. 

I also found that when I use the jack for snapcast, the system can not use the jack to output sound from desktop. So there seems to be some conflict between the desktop sound and snapclient. 

Now I have something to work from.

**Comment by PD75** [comment]
Some progress, now I need some advice on how to go forward. If I mute the snapclient then I can output sound from the desktop through the jack. And when I stop the sound from desktop, I can resume the sound from snapclient. Same behaviour as on Pi4, so it seems to be running through ALSA. This is a very clean approach as I can optimise sound quality for every device on its own with no pulse interference (well except for desktop).

I see two options forward:
Preferrable - running through ALSA to HDMI and then it will only work when pulse is quiet. Just like it does for the jack now.
Backup, dump the sound through Pulse, which means I could get sounds mixed, pulse resamples etc etc etc.

What is my next steps to force snapclient to one or the other?

**Comment by PD75** [comment]
So, I did some googling and found some previous questions here, I tried around with different outputs, and found one that works like a dream(much better then the Pi4), I got it to run cleanly now with snapclient and kodi through ALSA and the rest through Pulse Audio. So it gets exclusive. 

Next project is to play with sample rates on Snapclient. Considering the hardware I should be able to run Hi-Res sound through snapcast as well...
